st2common/st2common/models/api/keyvalue.py

```diff
@@ -188,7 +188,8 @@ def to_model(cls, kvp):
                 raise ValueError(msg)
 
             # Additional safety check to ensure that the value hasn't been decrypted
-            assert value == original_value
+            if not value == original_value:
```

Comment:
This comparison is weird.

```suggestion
            if value != original_value:
```